Sources/HummingbirdHTTP2/HTTP2Channel.swift

```diff
@@ -0,0 +1,109 @@
+        self.http1 = HTTP1Channel(additionalChannelHandlers: additionalChannelHandlers(), responder: responder)
+        self.responder = responder
+    }
+
+    public func initialize(channel: Channel, configuration: HBServerConfiguration, logger: Logger) -> EventLoopFuture<Value> {
+        do {
+            try channel.pipeline.syncOperations.addHandler(NIOSSLServerHandler(context: sslContext))
+        } catch {
+            return channel.eventLoop.makeFailedFuture(error)
+        }
+
+        let childChannelHandlers: [RemovableChannelHandler] = self.additionalChannelHandlers() + [
+            HBHTTPUserEventHandler(logger: logger),
+            HBHTTPSendableResponseChannelHandler(),
+        ]
+        
+        return channel.configureAsyncHTTPServerPipeline { http1Channel -> EventLoopFuture<HTTP1Channel.Value> in
+            http1Channel
+                .pipeline
+                .addHandlers(childChannelHandlers)
+                .flatMapThrowing {
+                    try HTTP1Channel.Value(wrappingChannelSynchronously: http1Channel)
+                }
+        } http2ConnectionInitializer: { http2Channel -> EventLoopFuture<NIOAsyncChannel<HTTP2Frame, HTTP2Frame>> in
+            http2Channel.eventLoop.makeCompletedFuture {
+                try NIOAsyncChannel<HTTP2Frame, HTTP2Frame>(wrappingChannelSynchronously: http2Channel)
+            }
+        } http2StreamInitializer: { http2ChildChannel -> EventLoopFuture<HTTP1Channel.Value> in
+            http2ChildChannel
+                .pipeline
+                .addHandler(HTTP2FramePayloadToHTTP1ServerCodec())
+                .flatMap {
+                    http2ChildChannel.pipeline.addHandlers(childChannelHandlers)
+                }.flatMapThrowing {
+                    try HTTP1Channel.Value(wrappingChannelSynchronously: http2ChildChannel)
+                }
+        }
+    }
+
+    public func handle(value: Value, logger: Logger) async {
+        do {
+            let channel = try await value.get()
+            switch channel {
+            case .http1_1(let http1):
+                await handleHTTP(asyncChannel: http1, logger: logger)
+            case .http2((let http2, let multiplexer)):
+                try await withThrowingDiscardingTaskGroup { group in
+                    for try await client in multiplexer.inbound {
+                        group.addTask {
+                            await handleHTTP(asyncChannel: client, logger: logger)
+                        }
+                    }
+                }
+
+                // Close `http2` here, because we're not interested in observing inbound/outbound here
+                // Closing it means we retain the `http2.channel`, ensuring it doesn't get deallocated
+                // until the `multiplexer` is done.
+                // Not having this will cause HTTP2 connections to close shortly after the first request
+                // is handled.
+                try await http2.channel.close()
```

Comment:
Your comment doesn't make sense here. Are you saying that calling a function we won't reach until all HTTP2 streams are closed stops the HTTP2 streams closing.

By the way I still think it is a good idea to close the http2 channel once we are done with it.

Reply:
That's exactly what happens, because if we don't refer to `http2`, its `deinit` closes the channel

Reply:
Had me scratching my head for two days

Reply:
Can you make it clear in the comment the reason it works is because you are referencing the http type. This still confuses me a little seeing as its owning type channel is still in scope. 

I think possibly the issue you had may be related to how you had everything structured previously and were adding everything to the one discarding task group. And it is no longer an issue.

Reply:
I've updated it, please review.